TraderJoe-Bots/Re-entrancy-Transfers/src/agent.spec.ts

```diff
@@ -0,0 +1,400 @@
+import { BigNumber, Transaction } from "ethers";
+import { Interface } from "ethers/lib/utils";
+import {
+  FindingType,
+  FindingSeverity,
+  Finding,
+  HandleTransaction,
+  createTransactionEvent,
+  TransactionEvent,
+  Trace,
+  TraceAction,
+} from "forta-agent";
+import { MockEthersProvider } from "forta-agent-tools/lib/mock.utils";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests.utils";
+import { provideHandleTransaction } from "./agent";
+import MarketsFetcher from "./markets.fetcher";
+import NetworkData from "./network";
+import { FUNCTIONS_IFACE, FUNCTIONS_MAP, MARKET_UPDATE_ABIS } from "./utils";
+
+describe("Re-entrancy transfers bot test suite", () => {
+  let handleTransaction: HandleTransaction;
+
+  const USER_ADDR = createAddress("0xc1");
+  const TEST_MARKETS = [createAddress("0x11"), createAddress("0x22"), createAddress("0x33")];
+
+  const mockNetworkManager: NetworkData = {
+    joeTroller: createAddress("0x0ae"),
+    sJoeStaking: createAddress("0x1ae"),
+    masterChefV2: createAddress("0x2ae"),
+    moneyMaker: createAddress("0x3ae"),
+    networkMap: {},
+    setNetwork: jest.fn(),
+  };
+  const mockProvider = new MockEthersProvider();
+  const mockFetcher = new MarketsFetcher(mockProvider as any);
+  const mockUpdateMarkets = jest.fn();
+
+  const createTrace = (from: string, traceAddress: number[], to: string, input: string): Trace => {
+    return {
+      action: {
+        from,
+        to,
+        input,
+      } as TraceAction,
+      traceAddress,
+    } as Trace;
+  };
+
+  const createFinding = (
+    from: string,
+    address: string,
+    rentrancyFrom: string,
+    initialCall: string,
+    reEtrantCall: string
+  ) => {
+    let reEtrantSig;
+    try {
+      reEtrantSig = FUNCTIONS_MAP.get(reEtrantCall);
+    } catch {
+      // if the reentrant call is different from the monitored transfer calls, return the selector.
+      reEtrantSig = reEtrantCall;
+    }
+
+    return Finding.fromObject({
+      name: "Re-entrancy detected on a Trader Joe contract",
+      description: "A function call to a Trader Joe contract resulted in another call to the same contract",
+      alertId: "TRADERJOE-25",
+      protocol: "TraderJoe",
+      severity: FindingSeverity.High,
+      type: FindingType.Exploit,
+      metadata: {
+        from: from, // address initializing the call.
+        initialCall: FUNCTIONS_MAP.get(initialCall)?.slice(9) as string, // function that was called in our contract and resulted in a re-entrancy.
+        entrancyFrom: rentrancyFrom, // the contract re-entring our contract.
```

Comment:
```suggestion
        entrancyFrom: rentrancyFrom, // the contract re-entering our contract.
```
please check for typos related to reentrancy.

Additionally I will not say `our`